util/types/datum.go

```diff
@@ -950,6 +1014,9 @@ func (d *Datum) convertToMysqlTime(sc *variable.StatementContext, target *FieldT
 		t   Time
 		err error
 	)
+	if tp == mysql.TypeTimestamp {
```

Comment:
Why do we need to add this judgment? If needs to add it here, why other functions of `convertToxxx` don't need to.